### DISCUSSION TITLE
opencart integration issue: php unknown, no payment options found

### DISCUSSION THREAD

**Discussion by littlebear1123** [btcpayserver/btcpayserver#5942] [Technical Support]
open cart error log: PHP Unknown:  Creation of dynamic property Template\Twig::$registry is deprecated .....
when check out ,it shows no payment option found.
opencart vision 3, 
lastest vision Btcpay serve

Can anyone help me?

<img width="613" alt="Screenshot 2024-04-22 at 10 23 58 PM" src="https://github.com/btcpayserver/btcpayserver/assets/163215813/34fbe63a-ea08-4fa6-86ae-e0a31895c756">
<img width="983" alt="Screenshot 2024-04-22 at 10 05 30 PM" src="https://github.com/btcpayserver/btcpayserver/assets/163215813/c0950be9-e280-4843-ba24-0c3e9f23647e">

**Comment by pavlenex** [comment]
cc @ndeet any idea what's happening here?

**Reply by ndeet** [reply]
@littlebear1123 did you enable the payment method? see 4. here:
https://docs.btcpayserver.org/OpenCart/#21-configure-btcpay-server-extension-in-opencart

The twig error seems to come from opencart itself not from BTCPay extension. If you enabled and saved the payment method and the problem persists can you please give me the exact OpenCart version and PHP version? You can see it in the blue box when you are editing the BTCPay payment option.

**Comment by littlebear1123** [comment]
Hi,

Thank you for responding. Here are the information: Debug info: OpenCart
3.0.3.9 with BTCPay Extension 3.1.0 on PHP 8.3.6
I really appreciate your help.

Littlebear

**Reply by ndeet** [reply]
Hi, 

thank you, can you please double check on the [link from above](https://docs.btcpayserver.org/OpenCart/#21-configure-btcpay-server-extension-in-opencart) section 4., so that on the extension settings page if BTCPay payment method is enabled?

**Reply by ndeet** [reply]
OpenCart version 3 is quite old, I doubt it works with PHP 8.3.x you likely need to downgrade to PHP 8.0 or even PHP 7.4, see "Compatible PHP versions" here https://next-cart.com/blog/opencart-4-vs-opencart-3/

**Comment by littlebear1123** [comment]
Hi,

I'm sure it is enabled. According to the link you sent me , it said Open
cart 3.0.3.8 is the most stable vision so far. I really appreciate your
information. Could you guide me ro downgrade to PHP 8.0 or even PHP 7.4? If
it still doest wokr, i will consider updating open cart to the
latest version. Have a lovely day !

Littlebear

**Reply by ndeet** [reply]
Downgrading your PHP version is done over the hosting provider usually. As you have the most recent version PHP 8.3 it seems they are a good one and likely offer to switch PHP versions. The docs or faq of your hosting provider should give you more info on this.

OpenCart 3 is not the latest one, there is OpenCart 4.0.2.3 which supports PHP 8.1 and newer. So if you just started out you maybe want to try to get started with OpenCart 4.
